komposable-architecture/src/test/java/com/toggl/komposable/effect/EffectCancellationTests.kt

```diff
@@ -0,0 +1,177 @@
+package com.toggl.komposable.effect
+
+import app.cash.turbine.turbineScope
+import com.toggl.komposable.architecture.Effect
+import com.toggl.komposable.extensions.cancel
+import com.toggl.komposable.extensions.cancellable
+import com.toggl.komposable.extensions.merge
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.types.shouldBeInstanceOf
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Test
+import kotlin.coroutines.cancellation.CancellationException
+
+class EffectCancellationTests {
+    @Test
+    fun `effects can be canceled with cancel effect`() = runTest {
+        val effect1Flow = MutableStateFlow(0)
+        val effect1 = Effect {
+            effect1Flow
+        }.cancellable("effect1")
+
+        val effect2Flow = MutableStateFlow(0)
+        val effect2 = Effect {
+            effect2Flow
+        }.cancellable("effect2")
+
+        val cancelEffect1 = Effect.cancel("effect1")
+
+        turbineScope {
+            val effect1Turbine = effect1().testIn(backgroundScope)
+            val effect2Turbine = effect2().testIn(backgroundScope)
+
+            effect1Turbine.awaitItem() shouldBe 0
+            effect1Flow.value = 1
+            effect1Turbine.awaitItem() shouldBe 1
+
+            effect2Turbine.awaitItem() shouldBe 0
+            effect2Flow.value = 1
+            effect2Turbine.awaitItem() shouldBe 1
+
+            cancelEffect1().testIn(backgroundScope)
+            effect1Turbine.awaitError().shouldBeInstanceOf<CancellationException>()
+            // effect2 should not be canceled
+            effect2Flow.value = 2
+            effect2Turbine.awaitItem() shouldBe 2
+        }
+    }
+
+    @Test
+    fun `multiple effects with the same id should all be canceled at the same time`() = runTest {
+        val effect1Flow = MutableStateFlow(0)
+        val effect1 = Effect {
+            effect1Flow
+        }.cancellable("effect", cancelInFlight = false)
+
+        val effect2Flow = MutableStateFlow(0)
+        val effect2 = Effect {
+            effect2Flow
+        }.cancellable("effect", cancelInFlight = false)
+
+        val cancelEffect = Effect.cancel("effect")
```

Comment:
If the runner does test paralellization the right way, the fact that this test is sharing an effect id with a different test can cause weird errors. Maybe using unique effect ids in the tests is a good way of showing good practices